proto/messages.go

```diff
@@ -1198,10 +1198,7 @@ func ReadVersionedFetchResp(r io.Reader, version int16) (*FetchResp, error) {
 
 	resp.Version = version
 
-	//Have to wrap because we use Peek later
-	br := bufio.NewReader(r)
-	dec := NewDecoder(br)
-	r = nil // just to prevent later usage
+	dec := NewDecoder(r)
```

Comment:
Are you sure that it is safe?  I do remember that there are some edge cases with Decoder.  Because it uses an internal buffer and can read to this buffer ahead of time.

Reply:
I'll do an audit, but I'm pretty sure I've seen other cases where we use a decoder for a while and then just go back to the underlying stream.

Reply:
So, a couple of points:
- I read through `decoder` and I don't see anywhere where it reads ahead.
- If it did, I think we'd have bigger issues, because it doesn't use `Peek` on the underlying stream, so it would never be safe to use the stream after `decoder` touches it, but we were already doing that

Reply:
You are right. I don't remember details, I just have some subconscious anxiety, because I remember that I caught some nasty bug related to incorrect reader usage.